### DISCUSSION TITLE
Union of types as attribute type in strawberry input class

### DISCUSSION THREAD

**Discussion by eddysen** [strawberry-graphql/strawberry#2420] [General] [answered]
Hello, can any one show me how i can manage to use Union of types as type for attribute in strawberry input class ?
I need to do something like this : 
```
@strawberry.input
class DesignationClientInType:
	designation : Union[ProduitClientInType, KitClientInType]
```
when trying it's telling me that Union for ProduitClientInType is not supported because it is an Input type

**Comment by patrick91** [comment, marked as answer]
Hi @eddysen! Unfortunately, this is not supported by the GraphQL spec :(

There's a potential alternative which is called `oneOf` but that hasn't been implemented yet in the spec, but it is easy to implement by hand: https://stepzen.com/blog/coming-soon-to-graphql-the-oneof-input-object

maybe we can implement it in future if there's enough demand 😊

**Reply by eddysen** [reply]
Hello @patrick91 ,
Thanks for the quick answer. Hope you can implement it in future. it will be very helpful. Is there in strawberry input class any alternative way to do it right now ?

**Reply by patrick91** [reply]
unfortunately not, I'll open an issue about implementing oneOf 😊

**Reply by eddysen** [reply]
Okay, i see. Thanks.
